internal/aad/jwt.go

```diff
@@ -107,6 +125,15 @@ func JWTProviderWithManagedIdentity(clientID, environment string) JWTProviderOpt
 	}
 }
 
+// JWTProviderWithManagedIdentity configures the TokenProvider using managed identity
+// - attempt to authenticate using managed identity. If clientID is supplied then it will use that
+//   as the clientID of the user assigned managed identity. Otherwise it will default to the system assigned managed
+//   identity
+// Deprecated use JWTProviderWithManagedIdentityClientID
+func JWTProviderWithManagedIdentity(clientID, environment string) JWTProviderOption {
```

Comment:
Deprecated the unspecific signature. use `JWTProviderWithManagedIdentityClientID` or `JWTProviderWithManagedIdentityResourceID` instead